File: snake-ladder-game/src/main/java/com/debanjanc/snake/and/ladder/simulator/domain/Game.java
package com.debanjanc.snake.and.ladder.simulator.domain;

import java.util.Collection;

public abstract class Game {

	private final Board board;
	private final Collection<Player> players;
	private final Dice dice;
	private final Integer diceRollPerUser;

	public Game(Board board, Collection<Player> players, Dice dice, Integer diceRollPerUser) {
		this.board = board;
		this.players = players;
		this.dice = dice;
		this.diceRollPerUser = diceRollPerUser;
	}

	public Board getBoard() {
		return board;
	}

	public Collection<Player> getPlayers() {
		return players;
	}

	public Dice getDice() {
		return dice;
	}

	public Integer getDiceRollPerUser() {
		return diceRollPerUser;
	}

}

File: snakeAndLadder/SnakeAndLadder/game.cpp
#pragma once
#include<bits/stdc++.h>
#include "./Models/dice.cpp"
#include "./Models/ladder.cpp"
#include "./Models/snake.cpp"
#include "./Models/player.cpp"
#include "./Models/board.cpp"

using namespace std;

class Game{
    public:
        Board *board;
        bool mainLoop;

    Game(Board *board){
        mainLoop = true;
        this->board = board;
    }

    void start(){
        while(mainLoop)
            mainLoop = board->move();
    }
};
